src/UsgsAstroFramePlugin.cpp

```diff
@@ -277,207 +273,199 @@ csm::Model *UsgsAstroFramePlugin::constructModelFromISD(const csm::Isd &imageSup
-  sensorModel->m_iTransL[2] = atof(imageSupportData.param("itrans_line", 2).c_str());
-  if (imageSupportData.param("itrans_line", 0) == "") {
-    missingKeywords.push_back("itrans_line needs 3 elements");
-  }
-  else if (imageSupportData.param("itrans_line", 1) == "") {
-    missingKeywords.push_back("itrans_line needs 3 elements");
-  }
-  else if (imageSupportData.param("itrans_line", 2) == "") {
-    missingKeywords.push_back("itrans_line needs 3 elements");
-  }
-
-  sensorModel->m_nLines = atoi(imageSupportData.param("nlines").c_str());
-  sensorModel->m_nSamples = atoi(imageSupportData.param("nsamples").c_str());
-  if (imageSupportData.param("nlines") == "") {
-    missingKeywords.push_back("nlines");
-  }
-  if (imageSupportData.param("nsamples") == "") {
-    missingKeywords.push_back("nsamples");
-  }
-
-  sensorModel->m_transY[0] = atof(imageSupportData.param("transy", 0).c_str());
-  sensorModel->m_transY[1] = atof(imageSupportData.param("transy", 1).c_str());
-  sensorModel->m_transY[2] = atof(imageSupportData.param("transy", 2).c_str());
-  if (imageSupportData.param("transy", 0) == "") {
-    missingKeywords.push_back("transy 0");
-  }
-  else if (imageSupportData.param("transy", 1) == "") {
-    missingKeywords.push_back("transy 1");
-  }
-  else if (imageSupportData.param("transy", 2) == "") {
-    missingKeywords.push_back("transy 2");
-  }
-
-  sensorModel->m_transX[0] = atof(imageSupportData.param("transx", 0).c_str());
-  sensorModel->m_transX[1] = atof(imageSupportData.param("transx", 1).c_str());
-  sensorModel->m_transX[2] = atof(imageSupportData.param("transx", 2).c_str());
-  if (imageSupportData.param("transx", 0) == "") {
-    missingKeywords.push_back("transx 0");
-  }
-  else if (imageSupportData.param("transx", 1) == "") {
-    missingKeywords.push_back("transx 1");
-  }
-  else if (imageSupportData.param("transx", 2) == "") {
-    missingKeywords.push_back("transx");
-  }
-
-  sensorModel->m_majorAxis = 1000 * atof(imageSupportData.param("semi_major_axis").c_str());
-  if (imageSupportData.param("semi_major_axis") == "") {
-    missingKeywords.push_back("semi_major_axis");
+  sensorModel->m_currentParameterValue[3] = atof(imageSupportData.param("sensor_orientation", 0).c_str());
+  sensorModel->m_currentParameterValue[4] = atof(imageSupportData.param("sensor_orientation", 1).c_str());
+  sensorModel->m_currentParameterValue[5] = atof(imageSupportData.param("sensor_orientation", 2).c_str());
+  sensorModel->m_currentParameterValue[6] = atof(imageSupportData.param("sensor_orientation", 3).c_str());
+
+  if (imageSupportData.param("sensor_orientation", 0) == "") {
+    missingKeywords.push_back("w");
+  }
+  if (imageSupportData.param("sensor_orientation", 1)== "") {
+    missingKeywords.push_back("v1");
+  }
+  if (imageSupportData.param("sensor_orientation", 2) == "") {
+    missingKeywords.push_back("v2");
+  }
+  if (imageSupportData.param("sensor_orientation", 3) == "") {
+    missingKeywords.push_back("v3");
```

Comment:
Our warnings about what failed in ISD construction should use the keywords from the ISD spec. We should either report an error with "sensor_orientation" or make the quaternions an object so that "w", "v1", "v2", and "v3" are meaningful to users.